meilisearch-http/src/routes/search.rs

```diff
@@ -105,8 +105,13 @@ impl SearchQuery {
             .schema(&reader)?
             .ok_or(Error::internal("Impossible to retrieve the schema"))?;
 
-        let mut search_builder = index.new_search(self.q.clone());
+        let query = self
+            .q
+            .clone()
+            .map(|q| if q.is_empty() { None } else { Some(q) })
+            .flatten();
```

Comment:
https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then

Reply:
Thanks, it's fixed!